artix7/primitives/slicel/slicel.pb_type.xml

```diff
@@ -141,30 +169,116 @@
     <!-- [A-D]MUX -->
     <mux name="DMUX"
       input="BLK_IG-D5FF.Q BLK_IG-CARRY[3].O BLK_IG-CARRY[3].CO BLK_IG-DLUT.O6 BLK_IG-DLUT.O5"
-      output="BLK_IG-SLICEL.DMUX" />
+      output="BLK_IG-SLICEL.DMUX">
+      <metadata>
+        <meta name="fasm_mux">
+          BLK_IG-D5FF.Q => DMUX.D5Q
+          BLK_IG-DLUT.O5 => DMUX.O5
+          BLK_IG-DLUT.O6 => DMUX.O6
+          BLK_IG-CARRY[3].CO => DMUX.CY
+          BLK_IG-CARRY[3].O => DMUX.XOR
+        </meta>
```

Comment:
No, if BLK_IG-DFF.Q path enable, then connect input of mux DMUX.D5Q to output of mux

Reply:
I would like to avoid encoding "another" format inside XML. We already have XML and the weird port/bit syntax plus array syntax.

The metadata stuff seems to already support the concept of a value at an offset on the rr_nodes. Could we reuse that?

If I recall correctly,
```xml
 <mux name="bla" input="multibit.port" output="blah">
```
Is the same as 
```xml
 <mux name="bla" input="multibit.port[0] multibit.port[1] ... Multibit.port[n]" output="blah">
```

Reply:
I still dislike the current syntax, it's encoding another format inside an XML document.

Reply:
That's actually true of all the FASM metadata, so I'm not using that as a bar to cross.  I don't think the LUT metadata should be XML e.g. `<lut start_bit="4" end_bit="6">LUT.INIT</lut>` versus LUT.INIT[6:4].

Reply:
FYI - `LUT.INIT[6:4]` is an existing format that can be used in a number of other places in the XML file.